server/controllers/stock/reports/stock_entry_donation.receipt.handlebars

```diff
@@ -0,0 +1,107 @@
+          <span class="text-uppercase">
+            {{translate 'STOCK.RECEIPT.ENTRY_DONATION'}}
+          </span> <br>
+          <strong>{{ details.depot_name}}</strong> <br>
+          <strong>{{ details.document_reference}}</strong> <br>
+          <small>{{date details.date}}</small> <br>
+          <br>
+        </h3>
+      </div>
+    </div>
+
+    <!-- client and user details  -->
+    <div class="row" style="border: 1px solid #ccc; padding: 5px; margin-bottom: 15px;">
+      <div class="col-xs-6">
+        <h4>{{translate 'STOCK.FROM'}}: {{translate 'STOCK.ENTRY_DONATION'}}</h4>
+        <span class="text-capitalize">{{translate 'FORM.LABELS.DATE'}}</span>: {{date details.date}} <br>
+      </div>
+      <div class="col-xs-6">
+        <h4>{{translate 'STOCK.TO'}}</h4>
+        <span class="text-capitalize">{{translate 'STOCK.DEPOT'}}</span>: <strong>{{details.depot_name}}</strong> <br>
+        <span class="text-capitalize">{{translate 'FORM.LABELS.DOCUMENT'}}</span>: <strong>{{details.document_reference}}</strong> <br>
+        <span class="text-capitalize">{{translate 'FORM.LABELS.DATE'}}</span>: {{date details.date}} <br>
+        <span class="text-capitalize">{{translate 'FORM.LABELS.COST'}}</span>: {{currency (sum rows 'total') enterprise.currency_id}} <br>
+        <span class="text-capitalize">{{translate 'STOCK.INVENTORY'}}</span>: {{rows.length}} {{translate 'STOCK.ITEMS'}} <br>
+        <span class="text-capitalize">{{translate "TABLE.COLUMNS.CREATED_BY"}}</span>: {{details.user_display_name}} <br>
+      </div>
+    </div>
+  </header>
+
+  {{#if details.description}}
+  <blockquote>
+    <strong>{{translate 'FORM.LABELS.DESCRIPTION'}}</strong><br>
+    {{details.description}}
+  </blockquote>
+  {{/if}}
+
+  <!-- list of data  -->
+  <table class="table table-condensed table-bordered table-striped">
+    <thead>
+      <tr>
+        <th>{{translate 'STOCK.CODE'}}</th>
+        <th>{{translate 'STOCK.INVENTORY'}}</th>
+        <th>{{translate 'STOCK.LOT'}}</th>
+        <th>{{translate 'STOCK.EXPIRATION'}}</th>
+        <th>{{translate 'STOCK.QUANTITY'}}</th>
+        <th>{{translate 'STOCK.UNIT_COST'}}</th>
+        <th>{{translate 'TABLE.COLUMNS.TOTAL'}}</th>
+      </tr>
+    </thead>
+    <tbody>
+      {{#each rows}}
+        <tr>
+          <td>{{code}}</td>
+          <td>{{text}}</td>
+          <td>{{label}}</td>
+          <td>{{date expiration_date}}</td>
+          <td class="text-right">{{quantity}}</td>
+          <td class="text-right">{{currency unit_cost ../enterprise.currency_id}}</td>
+          <td class="text-right">{{currency total ../enterprise.currency_id}}</td>
+        </tr>
+      {{else}}
+        {{> emptyTable columns=9}}
+      {{/each}}
+    </tbody>
+    <tfoot>
+      <tr style="font-weight: bold;">
+        <td colspan="6">{{rows.length}} {{translate 'STOCK.ITEMS'}}</td>
+        <td class="text-right">{{currency (sum rows 'total') enterprise.currency_id}}</td>
```

Comment:
While this looks really cool, it's hard to optimize.  The best place to do any addition is in SQL, although JS would be acceptable for a first implementation.  Please move this logic to either SQL or JS.

For reference, see the following issues: https://github.com/IMA-WorldHealth/bhima-2.X/issues/1668, https://github.com/IMA-WorldHealth/bhima-2.X/issues/1145, https://github.com/Vanga-Hospital/bhima-2.X/issues/204

Reply:
👍